src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala

```diff
@@ -27,37 +26,42 @@ class FFIDMetadataService(
 
   val loggingUtils: LoggingUtils = LoggingUtils(Logger("FFIDMetadataService"))
 
-  def addFFIDMetadata(ffidMetadata: FFIDMetadataInput): Future[FFIDMetadata] = {
-
-    if (ffidMetadata.matches.isEmpty) {
-      throw InputDataException(s"No ffid matches for file ${ffidMetadata.fileId}")
-    }
-
-    val metadataRow = FfidmetadataRow(
-      uuidSource.uuid,
-      ffidMetadata.fileId,
-      ffidMetadata.software,
-      ffidMetadata.softwareVersion,
-      Timestamp.from(timeSource.now),
-      ffidMetadata.binarySignatureFileVersion,
-      ffidMetadata.containerSignatureFileVersion,
-      ffidMetadata.method
-    )
+  @deprecated("Use addFFIDMetadata(ffidMetadataInput: FFIDMetadataInput)")
+  def addFFIDMetadata(ffidMetadataInputValues: FFIDMetadataInputValues): Future[FFIDMetadata] = {
+    addFFIDMetadata(FFIDMetadataInput(ffidMetadataInputValues :: Nil)).map(_.head)
+  }
 
-    def addFFIDMetadataMatches(ffidmetadataid: UUID): Future[Seq[Tables.FfidmetadatamatchesRow]] = {
-      val matchRows = ffidMetadata.matches.map(m => FfidmetadatamatchesRow(ffidmetadataid, m.extension, m.identificationBasis, m.puid))
-      matchesRepository.addFFIDMetadataMatches(matchRows)
-    }
+  def addFFIDMetadata(ffidMetadataInput: FFIDMetadataInput): Future[List[FFIDMetadata]] = {
+    val (metadataRows, matchRows) = ffidMetadataInput.metadataInputValues
+      .map(ffidMetadata => {
+        if (ffidMetadata.matches.isEmpty) {
+          throw InputDataException(s"No ffid matches for file ${ffidMetadata.fileId}")
+        }
+
+        val metadataRow = FfidmetadataRow(
+          uuidSource.uuid,
+          ffidMetadata.fileId,
+          ffidMetadata.software,
+          ffidMetadata.softwareVersion,
+          Timestamp.from(timeSource.now),
+          ffidMetadata.binarySignatureFileVersion,
+          ffidMetadata.containerSignatureFileVersion,
+          ffidMetadata.method
+        )
+
+        val matchRows = ffidMetadata.matches.map(m => FfidmetadatamatchesRow(metadataRow.ffidmetadataid, m.extension, m.identificationBasis, m.puid))
+
+        (metadataRow, matchRows)
+      })
+      .unzip
 
-    (for {
-      fileStatusRows <- generateFileStatusRows(ffidMetadata)
-      _ = loggingUtils.logFileFormatStatus("FFID", ffidMetadata.fileId, fileStatusRows.map(_.value).mkString(","))
-      ffidMetadataRow <- ffidMetadataRepository.addFFIDMetadata(metadataRow, fileStatusRows)
-      ffidMetadataMatchesRow <- addFFIDMetadataMatches(ffidMetadataRow.ffidmetadataid)
+    for {
+      statuses <- Future.sequence(ffidMetadataInput.metadataInputValues.map(generateFileStatusRows))
```

Comment:
Is there a reason that we're no longer logging the statuses?

Reply:
I think I had some vague idea about replacing the Grafana monitoring that runs off these logs with something in the step functions but that's way off so I've put the logging back.